src/server.c

```diff
@@ -5253,6 +5960,39 @@ const char *getSafeInfoString(const char *s, size_t len, char **tmp) {
                        sizeof(unsafe_info_chars)-1);
 }
 
+sds genRedisInfoStringCommandStats(dict *commands) {
+    sds info = sdsempty();
```

Comment:
i think it is better to pass an input sds and accumulate to it, rather than generate an empty one and then copy it.